stonesoup/dataassociator/_assignment.py

```diff
@@ -317,3 +319,107 @@ def assign2DBasic(C):
 # SOFTWARE AND ANY RELATED MATERIALS, AND AGREES TO INDEMNIFY THE NAVAL
 # RESEARCH LABORATORY FOR ALL THIRD-PARTY CLAIMS RESULTING FROM THE ACTIONS
 # OF RECIPIENT IN THE USE OF THE SOFTWARE.
+
+
+def multidimensional_deconfliction(association_set):
+    """Solves the Multidimensional Assignment Problem (MAP)
+
+    The assignment problem becomes more complex when time is added as a dimension.
+    This basic solution finds all the conflicts in an association set and then creates a
+    matrix of sums of conflicts in seconds, which is then passed to assign2D to solve as a
+    simple 2D assignment problem.  Therefore, each object will only ever be assigned to one other
+    at any one time.  In the case of an association that only partially overlaps, the time range
+    of the "weaker" one (the one eliminated by assign2D) will be trimmed
+    until there is no conflict.
+
+    Due to the possibility of more than two conflicting associations at the same time,
+    this algorithm is recursive, but it is not expected many (if any) recursions will be required
+    for most uses.
+
+    Parameters
+    ----------
+    association_set: The :class:`AssociationSet` to de-conflict
+
+
+    Returns
+    -------
+    : :class:`AssociationSet`
+        The association set without contradictory associations
+    """
+    # Check if there are any conflicts.  If none we can simply return the input
+    if check_if_no_conflicts(association_set):
+        return association_set
+
+    objects = list(association_set.object_set)
+    length = len(objects)
+    totals = numpy.zeros((length, length))  # Time objects i and j are associated for in total
+
+    for association in association_set.associations:
+        if len(association.objects) != 2:
+            raise ValueError("Supplied set must only contain pairs of associated objects")
+        i, j = (objects.index(object_) for object_ in association.objects)
+        totals[i, j] = association.time_range.duration.total_seconds()
```

Comment:
Ok, so I *think*  = is correct. Because the AssociationSet will simplify together any associations describing the same pair of objects, and the associations are assumed to describe pairs, there will only only be one assignment of totals[i,j] at any level of recursion, which is desired I believe. Happy to discuss further if you're unsure.